tests/basic/mutex.rs

```diff
@@ -114,6 +116,125 @@ fn concurrent_increment() {
     });
 }
 
+#[test]
+/// Two concurrent threads trying to do an atomic increment using `try_lock`.
+/// One `try_lock` must succeed, while the other may or may not succeed.
+/// Thus we expect to see final values 1 and 2.
+fn concurrent_try_increment() {
+    let final_value_tracker = Arc::new(
```

Comment:
Feels a bit heavyweight (and harder to read) to use a HashMap here.  Maybe use a HashSet similar to what's done in the tests for atomics?

Reply:
Yeah, I think the HashSet approach might be a little clearer in both these tests.